File: app/src/main/java/com/cleanup/todoc/domaine/usecase/DeleteProjectUseCase.java
package com.cleanup.todoc.domaine.usecase;

import com.cleanup.todoc.data.repository.ProjectRepository;
import com.cleanup.todoc.domaine.ProjectDomainUiMapper;
import com.cleanup.todoc.domaine.model.ProjectDomain;
import com.cleanup.todoc.presentation.model.Project;

public class DeleteProjectUseCase {
    private final ProjectRepository repository;
    public DeleteProjectUseCase(ProjectRepository repository){this.repository=repository;}
    public void deleteProject(Project project){
        repository.delete(ProjectDomainUiMapper.mapToDomain(project));
    }
}

File: app/src/main/java/com/cleanup/todoc/domaine/usecase/DeleteProjectUseCase.kt
package com.cleanup.todoc.domaine.usecase

import com.cleanup.todoc.data.repository.ProjectRepository
import com.cleanup.todoc.domaine.ProjectDomainUiMapper
import com.cleanup.todoc.presentation.model.Project

class DeleteProjectUseCase(private val repository: ProjectRepository) {
    suspend operator fun invoke(project: Project) {
        repository.delete(ProjectDomainUiMapper.mapToDomain(project))
    }
}
